packages/runtime/src/widgets/Collapsible.tsx

```diff
@@ -71,9 +75,79 @@ export type CollapsibleProps = {
 } & EnsembleWidgetProps &
   HasItemTemplate;
 
+const EvaluateItem = React.memo(
+  ({
+    context,
+    template,
+    currentIndex,
+    setEvaluated,
+  }: {
+    context: object;
+    currentIndex: number;
+    template: CollapsibleItem;
+    setEvaluated: React.Dispatch<React.SetStateAction<CollapsibleTemplate[]>>;
+  }): null => {
+    const evaluatedData = useEvaluate({ ...template }, { context });
+
+    useEffect(() => {
+      const { key, label, children } = evaluatedData;
+
+      const header = isString(label)
+        ? label
+        : EnsembleRuntime.render([unwrapWidget(label)]);
+
+      const content = EnsembleRuntime.render([
+        unwrapWidget(children as string),
+      ]);
+
+      setEvaluated((prev) => {
+        const prevEvaluated = [...prev];
```

Comment:
Why duplicate the array?

Reply:
we need a shallow copy because React uses reference equality to detect changes. When you modify the array directly without creating a new reference, React assumes nothing has changed.